sdk/evm/tbrv3/contract.ts

```diff
@@ -171,20 +171,20 @@ export class Tbrv3 {
     }
 
     if (vaas.some(({payload}) => {
-      const destinationAddress = payload.to.address.toNative(payload.to.chain);
+      const destinationAddress = new EvmAddress(payload.to.address);
       // TODO: actually check that this is part of the supported EVM chains
-      return chainToPlatform(payload.to.chain) !== "Evm" || !destinationAddress.equals(toNative("Ethereum", this.address));
+      return chainToPlatform(payload.to.chain) !== "Evm" || !destinationAddress.equals(new EvmAddress(this.address));
     })) {
       // TODO: point out which one; do a search instead.
       throw new Error("At least one of the VAAs points to an unexpected contract.");
     }
 
     let value: bigint = 0n;
     for (const vaa of vaas) {
-      const tbrv3Message = layout.deserializeLayout(TBRv3Message, vaa.payload.payload);
+      const tbrv3Message = layout.deserializeLayout(versionedTbrMessageLayout, vaa.payload.payload);
```

Comment:
If this is how, we are deserializing nested payload, then we should also update relayer codebase too. I will do that.